server/src/routers/import/accounting.ts

```diff
@@ -31,17 +31,143 @@ const importAccountingSchema = {
+	}
+)
+
+router.patch("/profiles/:profile/payments",
+	upload.fields([{name: "payments"}]),
+	schema.validate(importAccountingSchema),
+	acl("profile-accounting", "write"),
+	async(req, res, next) => {
+		return createWorkerTask(req, res, FileType.PAYMENTS_FILE, true)
+	}
+)
+
+router.post("/profiles/:profile/events",
+	upload.fields([{name: "events"}]),
+	schema.validate(importAccountingSchema),
+	acl("profile-accounting", "write"),
+	async(req, res, next) => {
+		return createWorkerTask(req, res, FileType.EVENTS_FILE, false)
+	}
+)
+
+router.patch("/profiles/:profile/events",
+	upload.fields([{name: "events"}]),
+	schema.validate(importAccountingSchema),
+	acl("profile-accounting", "write"),
+	async(req, res, next) => {
+		return createWorkerTask(req, res, FileType.EVENTS_FILE, true)
+	}
+)
+router.post("/profiles/:profile/data",
+	upload.fields([{name: "data"}]),
+	schema.validate(importAccountingSchema),
+	acl("profile-accounting", "write"),
+	async(req, res, next) => {
+		return createWorkerTask(req, res, FileType.DATA_FILE, false)
+	}
+)
+
+router.patch("/profiles/:profile/data",
+	upload.fields([{name: "data"}]),
+	schema.validate(importAccountingSchema),
+	acl("profile-accounting", "write"),
+	async(req, res, next) => {
+		return createWorkerTask(req, res, FileType.DATA_FILE, true)
+	}
+)
+
+router.patch("/profiles/:profile/accounting",
+	upload.fields([{name: "accounting"}]),
+	schema.validate(importAccountingSchema),
+	acl("profile-accounting", "write"),
+	async(req, res, next) => {
+		return createWorkerTask(req, res, FileType.ACCOUNTING_FILE, true)
+	}
+)
+async function createWorkerTask(req, res, type: FileType, isAppend: boolean) {
+	// When file missing throw error immediately
+	if (!req.files || !req.files[type.toString()]) return res.status(400).send("Missing data file or zip file");
+	if (isNaN(req.body.year)) return res.status(400).send("Invalid year value");
+
+    // check if tokenCode in profile is same as in token. if not, the token has been revoked (revoke all current tokens by changing the code)
+	const profile = await db<ProfileRecord>("app.profiles").select("id", "tokenCode").where({ id: req.params.profile }).first();
+	if (req.user.tokenCode && req.user.tokenCode !== profile.tokenCode) return res.status(403).send("Token revoked.");
+
+    // check if imported year is created, if not create a new hidden year
+	var year: { profileId: number, year: number } = await db<YearRecord>("app.years")
+		.where({ profileId: req.params.profile, year: req.body.year })
+		.first();
+
+	if (!year) {
+		const yearInsert = await db<YearRecord>("app.years").insert({ profileId: Number(req.params.profile), year: req.body.year }, ["profileId", "year"]).then()
+		year = yearInsert ? yearInsert[0] : null;
+		if (!year) return res.status(500).send("Failed to create new accounting year in database.");
+	}
+
+    // add import task to database queue (worker checks the table)
+	var importData: Partial<ImportRecord> = {

```

Comment:
If I import several files at once. It seems that it creates one job per one file type? Is that the correct behavior. How does it handle the rollback in this scenario?

Reply:
Yes, if you upload (e.g) three files in the upload form, three separate jobs are created. Each job has it's own DB transaction, so the rollbacks are handled separately for each file. The exception is the upload via .zip, which runs in a single job, no matter the number of files in the .zip.

Reply:
maybe update the documentation.md to clarify this behavior to the users.